src/main/java/frc/robot/subsystems/AmpBar.java

```diff
@@ -1,84 +1,89 @@
-        rightMotor.setNeutralMode(NeutralMode.Coast);
+    public void setState(AmpBarStates state) {
 
-        pivotEncoder.reset();
+        if (robot.isClimbing()) {
+            state = AmpBarStates.OUT;
+        } else {
+            this.state = state;
+        }
     }
-  
-    public void checkFaults() {
-        SmartDashboard.putBoolean("Left Amp Bar Working", leftMotor.getTemperature() > 1); // returns 0 if no signal 
-        SmartDashboard.putBoolean("Right Amp Bar Working", rightMotor.getTemperature() > 1);
+
+    public boolean atSetPoint() {
+        double motorPosition = pivotEncoder.getPosition();
+        return Math.abs(motorPosition - pivotMotorSetpoint) <= Constants.AmpBar.ERROR_OF_MARGIN;
     }
 
     public void periodic() {
-        if (state == AmpBarStates.OUT) {
+        if (state == AmpBarStates.SHOOTING) {
             pivotMotorSetpoint = Constants.AmpBar.OUT;
-            stateString = "Amp Bar Out";
+            wheelsMotor.set(Constants.AmpBar.WHEEL_SPEED);
+            stateString = "Shooting Amp";
+
         } else if (state == AmpBarStates.IN) {
             pivotMotorSetpoint = Constants.AmpBar.IN;
+            wheelsMotor.set(0);
             stateString = "Amp Bar In";
+        } else if (state == AmpBarStates.OUT) {
+            pivotMotorSetpoint = Constants.AmpBar.IN;
+            wheelsMotor.set(0);
+            stateString = "Amp Bar Out";
         }
-        if (pivotEncoder.getAbsolutePosition() > 0.1) {
-            leftMotor.set(pivotController.calculate(pivotEncoder.getAbsolutePosition(), pivotMotorSetpoint));
-        } else {
-            System.out.println("Amp Bar Encoder Unplugged zzzzz");
-            if (robot.secondaryController.getPOV() == Constants.DPAD_LEFT) {
-                leftMotor.set(-0.2);
-            } else if (robot.secondaryController.getPOV() == Constants.DPAD_RIGHT) {
-                leftMotor.set(0.2);
-            } else {
-                leftMotor.set(0);
-            }
-        }
-    }
 
-    public void putSmartDashValues() {
-        SmartDashboard.putNumber("Amp motor setpoint", pivotMotorSetpoint);
-        SmartDashboard.putNumber("Current Amp motor postition", pivotEncoder.getAbsolutePosition());
-        SmartDashboard.putString("Amp Bar State", stateString);
+        leftMotor.set(controller.calculate(pivotEncoder.getPosition(), pivotMotorSetpoint));
     }
 
-    public boolean nearSetpoint() {
-        return Math.abs(pivotEncoder.getAbsolutePosition() - pivotMotorSetpoint) < 0.1;
+    public void checkFaults() {
+        SmartDashboard.putBoolean("left ampbar motor good",
+                leftMotor.getMotorTemperature() > 1 && leftMotor.getFaults() == 0);
+        SmartDashboard.putBoolean("right ampbar motor good",
+                rightMotor.getMotorTemperature() > 1 && rightMotor.getFaults() == 0);
+        SmartDashboard.putBoolean("amp wheel motor good", wheelsMotor.getDeviceTemp().getValueAsDouble() > 0);
```

Comment:
Shouldn't you also be checking for TalonFX faults, instead of just checking to see if the motor is connected (via temp)?